RHDEVDOCS-6023: Installing Argo Rollouts CLI

Dhruv-Soni11 opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->

GitOps 1.13

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

https://issues.redhat.com/browse/RHDEVDOCS-6023

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

Argo Rollouts CLI preview link: https://76625--ocpdocs-pr.netlify.app/openshift-gitops/latest/argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.html#gitops-argo-rollouts-cli-overview_using-argo-rollouts-for-progressive-deployment-delivery

SME review: @jgwest 
QE review: @mehabhalodiya 
Internal peer review: @eromanova97
Peer review: 

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

jgwest commented on modules/gitops-argo-rolouts-cli-ovr.adoc:
```diff
@@ -0,0 +1,12 @@
+[id="gitops-argo-rollouts-cli-overview_{context}"]
+= Argo Rollouts CLI Overview
+
+Using the Argo Rollouts CLI plugin is optional because you can deploy and monitor an application by installing Argo Rollouts using the {gitops-title} Operator. Following the installation of Argo Rollouts, users have the choice to utilize the Argo Rollouts CLI plugin. This plugin effectively handles and displays Argo Rollouts resources directly from the command line interface.
```
>  Using the Argo Rollouts CLI plugin is optional because you can deploy and monitor an application by installing Argo Rollouts using the {gitops-title} Operator. Following the installation of Argo Rollouts, users have the choice to utilize the Argo Rollouts CLI plugin. This plugin effectively handles and displays Argo Rollouts resources directly from the command line interface.

What about:

The Argo Rollouts CLI plugin is not required to use Argo Rollouts, because you can instead deploy and monitor a Rollout resource using just the OpenShift Console or oc command. However, this plugin effectively handles and displays Argo Rollouts resources directly from the command line interface.

(The original text made it sound like the CLI plugin was involved in installing/monitoring the Argo Rollouts installation itself, which it is not.)

Dhruv-Soni11 replied:
Fixed.

jgwest commented on modules/gitops-argo-rolouts-cli-ovr.adoc:
```diff
@@ -0,0 +1,12 @@
+
+Using the Argo Rollouts CLI plugin is optional because you can deploy and monitor an application by installing Argo Rollouts using the {gitops-title} Operator. Following the installation of Argo Rollouts, users have the choice to utilize the Argo Rollouts CLI plugin. This plugin effectively handles and displays Argo Rollouts resources directly from the command line interface.
+
+The Argo Rollouts CLI plugin interacts with the Argo Rollouts controller and Argo Rollouts resources. The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, and terminate a failed Argo Rollouts deployment.
```
> The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, and terminate a failed Argo Rollouts deployment.

I would add this to the end:

The Argo Rollouts CLI plugin interacts with the Argo Rollouts controller and Argo Rollouts resources. The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, ~and~ terminate a failed Argo Rollouts deployment, _**and more**_.
(addition is _**bolded**_, and likewise see the ~strikethrough~)

(This makes it clear that the CLI has many many commands/use cases, and it is not just limited to those explicitly listed.)

Dhruv-Soni11 replied:
Fixed.

jgwest reviewed (changes requested):
Looks great, thanks @Dhruv-Soni11!

jgwest reviewed (approved):
LGTM!

Dhruv-Soni11 commented:
/retest

eromanova97 commented on modules/gitops-argo-rolouts-cli-ovr.adoc:
```diff
@@ -0,0 +1,12 @@
+[id="gitops-argo-rollouts-cli-overview_{context}"]
+= Argo Rollouts CLI Overview
+
+The Argo Rollouts CLI plugin is not required to use Argo Rollouts resources, because you can instead deploy and monitor a Rollout resource using the {OCP} Console or the `oc` command. However, usage of the Argo Rollouts CLI plugin allows you to handle Argo Rollouts resources and display it directly from the command line interface.
+
```
This does not seem to be too user-friendly to include why the user does not need it as the first information. This can be restructured so that we give benefits first, for example:

```suggestion
Argo Rollouts CLI is an optional plugin that allows you to handle and monitor Argo Rollouts resources directly through the command line interface instead of using {OCP} web console or the `oc` command. 
```
Also please keep in mind how to refer to objects: 
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-argo-rolouts-cli-ovr.adoc:
```diff
@@ -0,0 +1,12 @@
+
+The Argo Rollouts CLI plugin is not required to use Argo Rollouts resources, because you can instead deploy and monitor a Rollout resource using the {OCP} Console or the `oc` command. However, usage of the Argo Rollouts CLI plugin allows you to handle Argo Rollouts resources and display it directly from the command line interface.
+
+The Argo Rollouts CLI plugin interacts with the Argo Rollouts controller and Argo Rollouts resources. The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, terminate a failed Argo Rollouts deployment, and so on.
+
+The Argo Rollouts CLI plugin is an `oc/kubectl` plugin, and integrates directly with the `oc/kubectl` commands.
```
I do not find the first sentence useful to the user. It is better to include exactly what the user can do, instead of how the plugin operates (it is something that is typically done by engineers, but it is unnecessarry in minimalistic user-centric docs)

Also, this would look better if it was a list:
```suggestion
The Argo Rollouts CLI plugin enables you to perform many actions, including the following actions:

* Make changes to an Argo Rollouts image.
* Monitor the progress of an Argo Rollouts promotion.
* Proceed with the promotion steps in a canary deployment.
* Terminate a failed Argo Rollouts deployment.

The Argo Rollouts CLI plugin is an `oc/kubectl` plugin, and integrates directly with the `oc/kubectl` commands.
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-argo-rolouts-cli-ovr.adoc:
```diff
@@ -0,0 +1,12 @@
+
+The Argo Rollouts CLI plugin is not required to use Argo Rollouts resources, because you can instead deploy and monitor a Rollout resource using the {OCP} Console or the `oc` command. However, usage of the Argo Rollouts CLI plugin allows you to handle Argo Rollouts resources and display it directly from the command line interface.
+
+The Argo Rollouts CLI plugin interacts with the Argo Rollouts controller and Argo Rollouts resources. The Argo Rollouts CLI plugin enables you to make changes to an Argo Rollouts image, monitor the progress of an Argo Rollouts promotion, proceed with the promotion steps in a canary deployment, terminate a failed Argo Rollouts deployment, and so on.
+
+The Argo Rollouts CLI plugin is an `oc/kubectl` plugin, and integrates directly with the `oc/kubectl` commands.
```
`oc/kubectl`: Is this true for downstream as well? Or is it only `oc` commands for downstream?

Dhruv-Soni11 replied:
Let me check with the SME.

Dhruv-Soni11 replied:
I discussed it a bit more with @eromanova97 and I am keeping the term as "oc kubetcl".

eromanova97 commented on argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc:
```diff
@@ -36,6 +36,16 @@ include::modules/gitops-creating-rolloutmanager-custom-resource.adoc[leveloffset
+include::modules/gitops-argo-rolouts-cli-ovr.adoc[leveloffset=+1]
+
+// Installing Argo Rollouts CLI using Linux
+include::modules/gitops-installing-argo-rollouts-cli-using-linux.adoc[leveloffset=+1]
```
```suggestion
include::modules/gitops-installing-argo-rollouts-using-linux.adoc[leveloffset=+2]
```
ID must be the same as the file name: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#module-file-metadata

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-installing-argo-rollouts-using-linux_{context}"]
+= Installing Argo Rollouts CLI using Linux
```
```suggestion
= Installing Argo Rollouts CLI by using Linux
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+
+.Prerequisites
+
+* You have installed the {OCP} CLI (oc).
```
Just making sure, are these all the prereqs? What about creating a RolloutManager custom resource?

Dhruv-Soni11 replied:
Yes, this was the only pre-req required and added in the draft document attached in the JIRA ticket for this PR.

jgwest replied:
I can confirm RolloutManager resource is not required to install the CLI.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+[id="gitops-installing-argo-rollouts-using-linux_{context}"]
+= Installing Argo Rollouts CLI using Linux
+
+For Linux, you can install Argo Rollouts CLI by performing the following steps.
```
To improve the sentence a little bit: (I am not sure about using "following steps" if we first have prereqs:
```suggestion
If you are a Linux user, you can install Argo Rollouts CLI by performing the steps described in this procedure.
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+
+.Procedure
+
+. Download the latest version of the Argo Rollouts CLI binary, `kubectl-argo-rollouts`, by running the following command using `curl`.
```
```suggestion
. Download the latest version of the Argo Rollouts CLI binary, `kubectl-argo-rollouts`, by running the following command:
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+
+. Download the latest version of the Argo Rollouts CLI binary, `kubectl-argo-rollouts`, by running the following command using `curl`.
++
+.Example - Downloading the Argo Rollouts CLI binary
```
We don't use example title for commands: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output
```suggestion
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+. Download the latest version of the Argo Rollouts CLI binary, `kubectl-argo-rollouts`, by running the following command using `curl`.
++
+.Example - Downloading the Argo Rollouts CLI binary
+[source,yaml]
```
```suggestion
[source,terminal]
```

eromanova97 replied:
this is a terminal, not a YAML file

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+curl -LO https://github.com/argoproj/argo-rollouts/releases/latest/download/kubectl-argo-rollouts-linux-amd64
+----
+
+. Ensure that the `kubectl-argo-rollouts` binary is executable by running the following command.
```
```suggestion
. Ensure that the `kubectl-argo-rollouts` binary is executable by running the following command:
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+
+. Ensure that the `kubectl-argo-rollouts` binary is executable by running the following command.
++
+.Example - Marking the `kubectl-argo-rollouts` binary as executable
```
```suggestion

```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+. Ensure that the `kubectl-argo-rollouts` binary is executable by running the following command.
++
+.Example - Marking the `kubectl-argo-rollouts` binary as executable
+[source,yaml]
```
```suggestion
[source,terminal]
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Downloading the Argo Rollouts CLI binary
+[source,yaml]
+----
+curl -LO https://github.com/argoproj/argo-rollouts/releases/latest/download/kubectl-argo-rollouts-linux-amd64
```
Add Example output, if any

Dhruv-Soni11 replied:
Hi @eromanova97 - There is no explicit example output apart from the fact that if you run a ls -;lh it just state the name of the binary that is downloaded, i.e. kubectl-argo-rollouts-linux-amd64. Do you think that a one line statement that this binary package is downloaded would be fine?

eromanova97 replied:
Yes, that sounds great :+1:

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Marking the `kubectl-argo-rollouts` binary as executable
+[source,yaml]
+----
+chmod +x ./kubectl-argo-rollouts-linux-amd64
```
Add Example output

Dhruv-Soni11 replied:
No example output available.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+chmod +x ./kubectl-argo-rollouts-linux-amd64
+----
+
+. Move the `kubectl-argo-rollouts` binary to the system PATH by running the following command.
```
Why is the path capitalised here? 
```suggestion
. Move the `kubectl-argo-rollouts` binary to the system path by running the following command:
```

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+
+. Move the `kubectl-argo-rollouts` binary to the system PATH by running the following command.
++
+.Example - Moving the `kubectl-argo-rollouts` binary to the system path
```
```suggestion

```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Moving the `kubectl-argo-rollouts` binary to the system path
+[source,yaml]
+----
+sudo mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
```
Is this path always the same for every Linux user? or should it be a variable?

Dhruv-Soni11 replied:
Let me confirm this with the SME.

jgwest replied:
The short answer is yes, the path would be the same for all Linux users, and likewise this is the same [step as described in the upstream docs](https://argo-rollouts.readthedocs.io/en/stable/installation/#manual), implying it is sufficiently universal for all users.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+. Move the `kubectl-argo-rollouts` binary to the system PATH by running the following command.
++
+.Example - Moving the `kubectl-argo-rollouts` binary to the system path
+[source,yaml]
```
```suggestion
[source,terminal]
```

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+sudo mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
+----
+
+The Argo Rollouts CLI binary is installed.
```
```suggestion
```
This sentence is unnecesarry.

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+
+The Argo Rollouts CLI binary is installed.
+
+.Verification
+
+After you have completed the installation of the Argo Rollouts CLI plugin, verify the installation of the Argo Rollouts CLI plugin by running the following command:
```
```suggestion
. Verify that the pluging is installed correctly by running the following command:
```

Dhruv-Soni11 replied:
Fixed,

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+
+After you have completed the installation of the Argo Rollouts CLI plugin, verify the installation of the Argo Rollouts CLI plugin by running the following command:
+
+.Example - Verifying the Argo Rollouts version
```
```suggestion

```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+After you have completed the installation of the Argo Rollouts CLI plugin, verify the installation of the Argo Rollouts CLI plugin by running the following command:
+
+.Example - Verifying the Argo Rollouts version
+[source,yaml]
```
```suggestion
[source,terminal]
```

Dhruv-Soni11 replied:
Fixed.

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Downloading the Argo Rollouts CLI binary
+[source,yaml]
+----
+curl -LO https://github.com/argoproj/argo-rollouts/releases/latest/download/kubectl-argo-rollouts-linux-amd64
```
```suggestion
$ curl -LO https://github.com/argoproj/argo-rollouts/releases/latest/download/kubectl-argo-rollouts-linux-amd64
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Marking the `kubectl-argo-rollouts` binary as executable
+[source,yaml]
+----
+chmod +x ./kubectl-argo-rollouts-linux-amd64
```
```suggestion
$ chmod +x ./kubectl-argo-rollouts-linux-amd64
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Moving the `kubectl-argo-rollouts` binary to the system path
+[source,yaml]
+----
+sudo mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
```
```suggestion
$ sudo mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Moving the `kubectl-argo-rollouts` binary to the system path
+[source,yaml]
+----
+sudo mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
```
Also, is adding `sudo` here neccessarry? I would rather mention to the note that this command might have to be run with superuser priviledges.

Dhruv-Soni11 replied:
Let me check this with the SME.

Dhruv-Soni11 replied:
I discussed this further with the SME and will add a note stating that superuser privileges are required to run this command. Thanks @eromanova97

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Verifying the Argo Rollouts version
+[source,yaml]
+----
+oc argo rollouts version
```
```suggestion
$ oc argo rollouts version
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+oc argo rollouts version
+----
+
+As a result, the following output is displayed:
+
+.Example - Output of the Argo Rollouts version command
+[source,yaml]
```
```suggestion
+
.Example output
[source,terminal]
```

eromanova97 replied:
this is how you add exmaple outputs, see: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Verification
+
+After you have completed the installation of the Argo Rollouts CLI plugin, verify the installation of the Argo Rollouts CLI plugin by running the following command:
+
```
```suggestion
+
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+.Example - Output of the Argo Rollouts version command
+[source,yaml]
+----
+kubectl-argo-rollouts: v1.6.6+737ca89
+  BuildDate: 2024-02-13T15:39:31Z
+  GitCommit: 737ca89b42e4791e96e05b438c2b8540737a2a1a
+  GitTreeState: clean
+  GoVersion: go1.20.14
+  Compiler: gc
+  Platform: linux/amd64
```
how will the user determine that the installation worked correctly? Is there anything to look for in this output? Add callouts.

Dhruv-Soni11 replied:
As per my interaction with the SME, the installation is completed after the 3rd step in this procedure. The verification steps are added to check the version of the downloaded binary package.

jgwest replied:
Verification steps will also verify that the installation was completed correctly.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+  Compiler: gc
+  Platform: linux/amd64
+----
+
```
```suggestion
+
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-linux.adoc:
```diff
@@ -0,0 +1,69 @@
+  Platform: linux/amd64
+----
+
+[NOTE]
+====
+The output of the `oc argo rollouts version` command displays the build date, version, and platform of the Argo Rollouts binary.
+====
```
Why adding this as a note? This would be better to add as a callout or just a description following the output

Dhruv-Soni11 replied:
Added it in the callout created after the command execution.

eromanova97 commented on argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc:
```diff
@@ -36,6 +36,16 @@ include::modules/gitops-creating-rolloutmanager-custom-resource.adoc[leveloffset
+include::modules/gitops-installing-argo-rollouts-cli-using-linux.adoc[leveloffset=+1]
+
+// Installing Argo Rollouts CLI using Mac OS
+include::modules/gitops-installing-argo-rollouts-cli-using-mac-os.adoc[leveloffset=+1]
```
```suggestion
include::modules/gitops-installing-argo-rollouts-using-mac-os.adoc[leveloffset=+2]
```
Needs to correspond with ID

eromanova97 commented on argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc:
```diff
@@ -36,6 +36,16 @@ include::modules/gitops-creating-rolloutmanager-custom-resource.adoc[leveloffset
+include::modules/gitops-argo-rolouts-cli-ovr.adoc[leveloffset=+1]
+
+// Installing Argo Rollouts CLI using Linux
+include::modules/gitops-installing-argo-rollouts-cli-using-linux.adoc[leveloffset=+1]
```
Wrong nesting

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc:
```diff
@@ -36,6 +36,16 @@ include::modules/gitops-creating-rolloutmanager-custom-resource.adoc[leveloffset
+include::modules/gitops-installing-argo-rollouts-cli-using-linux.adoc[leveloffset=+1]
+
+// Installing Argo Rollouts CLI using Mac OS
+include::modules/gitops-installing-argo-rollouts-cli-using-mac-os.adoc[leveloffset=+1]
```
Wrong nesting

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-installing-argo-rollouts-using-mac-os_{context}"]
+= Installing Argo Rollouts CLI using Mac OS
```
```suggestion
= Installing Argo Rollouts CLI by using Mac OS
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+[id="gitops-installing-argo-rollouts-using-mac-os_{context}"]
+= Installing Argo Rollouts CLI using Mac OS
+
+For macOS, you can install the Argo Rollouts CLI by using the link:https://brew.sh[Homebrew] package manager.
```
```suggestion
If you are a macOS user, you can install the Argo Rollouts CLI by using the link:https://brew.sh[Homebrew] package manager.
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+
+.Prerequisites
+
+* You must have Homebrew (`brew`) installed.
```
Same comment to prereqs as with Linux module.

jgwest replied:
I can confirm this is the only prerequisite to install the CLI.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+
+.Procedure
+
+* Run the following command to install the Argo Rollouts CLI.
```
```suggestion
* Run the following command to install the Argo Rollouts CLI:
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
++
+[source,terminal]
+----
+$ brew install argoproj/tap/kubectl-argo-rollouts
```
Any verification steps here? Or command outputs?

Dhruv-Soni11 replied:
As per the walkthrough demo provided by the SME, we did not view any additional output after this step.

jgwest replied:
Correct, I based the structure of this section on https://docs.openshift.com/container-platform/4.15/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli-brew_cli-developer-commands

eromanova97 reviewed (commented):
Hello @Dhruv-Soni11 I finished my review, good job! For the future PRs I recommend looking into the OCP doc guidelines regarding the file names: (https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc) to ensure that the module metadata have every aspect correct, because it can be quite a hassle having to change the name of the file later on :+1:

Dhruv-Soni11 commented on argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc:
```diff
@@ -36,6 +36,16 @@ include::modules/gitops-creating-rolloutmanager-custom-resource.adoc[leveloffset
 include::modules/gitops-deleting-rolloutmanager-custom-resource.adoc[leveloffset=+1]
 
+// Argo Rollouts CLI Overview
+include::modules/gitops-argo-rolouts-cli-ovr.adoc[leveloffset=+1]
```
Fixed.

eromanova97 commented on argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc:
```diff
@@ -36,6 +36,15 @@ include::modules/gitops-creating-rolloutmanager-custom-resource.adoc[leveloffset
+// Argo Rollouts CLI overview
+include::modules/gitops-argo-rollouts-cli-overview.adoc[leveloffset=+1]
+
+// Installing Argo Rollouts CLI by using Linux
```
I am thinking to maybe call these chapters: `Installing Argo Rollouts CLI on Linux` and `Installing Argo Rollouts CLI on Mac OS` as it sounds better and is more concise.

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-argo-rollouts-cli-overview.adoc:
```diff
@@ -0,0 +1,17 @@
+* Proceed with the promotion steps in a canary deployment.
+* Terminate a failed Argo Rollouts deployment.
+
+The Argo Rollouts CLI plugin is an `oc kubectl` plugin, and integrates directly with the `oc kubectl` commands.
```
```suggestion
The Argo Rollouts CLI plugin directly integrates with `oc` and `kubectl` commands.
```

eromanova97 replied:
This makes it more concise while still conveying the same meaning :+1:

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
+
+.Prerequisites
+
+* You have installed the {OCP} CLI (oc).
```
```suggestion
* You have installed the {OCP} CLI (`oc`).
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
+$ chmod +x ./kubectl-argo-rollouts-linux-amd64
+----
+
+. Move the `kubectl-argo-rollouts` binary to the system path by running the following command.
```
```suggestion
. Move the `kubectl-argo-rollouts` binary to the system path by running the following command:
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
+
+[NOTE]
+====
+Ensure that you have `superuser` privileges to run this command.
```
```suggestion
Ensure that you have superuser privileges to run this command.
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
++
+[source,terminal]
+----
+$ sudo mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
```
```suggestion
# mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
```
Because of the following note, `sudo` does not need to be there. You can just use hash symbol to make this clear.

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
+Ensure that you have `superuser` privileges to run this command.
+====
+
+.Verification
```
```suggestion

```
GitOps does not use this format for verifications, so this line should not be included here.

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
+  Compiler: gc
+  Platform: linux/amd64 #<3>
+----
+<1> - Specifies the build date information of the Argo Rollouts binary
+<2> - Specifies the version of the `Go` language used for building the Argo Rollouts binary
+<3> - Specifies the platform used for building the Argo Rollouts binary
```
```suggestion
<1> The build date information of the Argo Rollouts binary.
<2> The version of the Go language used for building the Argo Rollouts binary.
<3> The platform used for building the Argo Rollouts binary.
```

Dhruv-Soni11 replied:
Fixed.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
+  Compiler: gc
+  Platform: linux/amd64 #<3>
+----
+<1> - Specifies the build date information of the Argo Rollouts binary
+<2> - Specifies the version of the `Go` language used for building the Argo Rollouts binary
+<3> - Specifies the platform used for building the Argo Rollouts binary
```
When I mentioned to add the callouts, I was mostly thinking to add callouts that would help user to determine what to look for here, because it is a verification step. The added callouts are good, I just want it to be clear how do we determine from this that the installation is correct.

Is just the fact that the output is there (the version command worked) enough of the verification? That is what I was trying to ask. Or if the user should look for the specific relevant information from the output that would help determine if the installation is correct.

eromanova97 replied:
If just the fact that the command worked is the verification step by itself, then I would just tweak the step sentence a little bit:
. Verify that the plugin is installed correctly by running the following command and receiving the similar output:

Dhruv-Soni11 replied:
Yeah, the idea is that this output is displayed after running this command is proof that the installation is successful. I will change the sentence slightly as per your suggestion @eromanova97. Thanks :)

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+
+.Prerequisites
+
+* You must have Homebrew (`brew`) installed.
```
```suggestion
* You have installed the Homebrew (`brew`) package manager.
```
To make it consistent with the previsous prerequisite.

eromanova97 commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,64 @@
+$ sudo mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
+----
+
+[NOTE]
```
```suggestion
[IMPORTANT]
```
This should probably be IMPORTANT admonition if it contains the word "ensure".

Dhruv-Soni11 replied:
Fixed.

eromanova97 reviewed (commented):
Hello @Dhruv-Soni11, I finished the review, I have some more small suggestions to improve it a little bit, but othervise it looks good! Thank you!

mehabhalodiya reviewed (approved):
LGTM,
Thanks!

Dhruv-Soni11 commented:
label peer-review-needed

Dhruv-Soni11 commented:
/label peer-review-needed

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+//
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-installing-argo-rollouts-cli-on-linux_{context}"]
```
File name and module ID should match.

Dhruv-Soni11 replied:
Fixed.

maxwelldb commented on modules/gitops-argo-rollouts-cli-overview.adoc:
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+:_mod-docs-content-type: CONCEPT
```
```suggestion

:_mod-docs-content-type: CONCEPT
```

Dhruv-Soni11 replied:
Fixed.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+:_mod-docs-content-type: PROCEDURE
```
```suggestion
// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc

:_mod-docs-content-type: PROCEDURE
```

Dhruv-Soni11 replied:
Fixed.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+:_mod-docs-content-type: PROCEDURE
```
```suggestion
// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc

:_mod-docs-content-type: PROCEDURE
```

Dhruv-Soni11 replied:
Fixed.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+//
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-installing-argo-rollouts-cli-on-mac-os_{context}"]
```
File name and module ID should match.

Dhruv-Soni11 replied:
Fixed.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+[id="gitops-installing-argo-rollouts-cli-on-linux_{context}"]
+= Installing Argo Rollouts CLI on Linux
+
+If you are a Linux user, you can install Argo Rollouts CLI by performing the steps described in this procedure.
```
Avoid self-referential language.

Maybe:

```suggestion
You can install the Argo Rollouts CLI on Linux.
```

Dhruv-Soni11 replied:
Fixed.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+
+.Prerequisites
+
+* You have installed the {OCP} CLI (`oc`)
```
Check out the `oc-first` common attribute.

Dhruv-Soni11 replied:
Fixed.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+# mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
+----
+
+[IMPORTANT]
+====
+Ensure that you have superuser privileges to run this command.
+====
```
Could work this into the previous command, or even as a step on its own prior to that. Is `sudo` an option?

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+Ensure that you have superuser privileges to run this command.
+====
+
+. Verify that the plugin is installed correctly by running the following command and receiving the similar output:
```
This could be the start of a `.Verification` section.

Dhruv-Soni11 replied:
This was a verification option earlier but as per the feedback received, I have added it into a step. Kindly advise on how to proceed.

maxwelldb replied:
I wasn't aware that GitOps didn't use that Verification style. It's a standard part of mod docs. 

https://redhat-documentation.github.io/modular-docs/#verification

If there's really a prohibition against it that is specific to GitOps, yeah, don't use it.

Dhruv-Soni11 replied:
No, as far as I know, there are no issues with using the verification style. It is just modified as per the review feedback received earlier.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+  Compiler: gc
+  Platform: linux/amd64 #<3>
+----
+<1> The build date information of the Argo Rollouts binary.
+<2> The version of the Go language used for building the Argo Rollouts binary.
+<3> The platform used for building the Argo Rollouts binary.
```
Is it important to point these out?

Dhruv-Soni11 replied:
Yes, it adds a bit of additional information for the end-user to refer if the installation is correct or not.

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-mac-os.adoc:
```diff
@@ -0,0 +1,21 @@
+[source,terminal]
+----
+$ brew install argoproj/tap/kubectl-argo-rollouts
+----
```
Is it worth including verification instructions beyond this point? The Linux module does that.

Dhruv-Soni11 replied:
I agree that the Linux module had some additional steps but during the walkthrough provided by the SME for this module, we did not have additional steps after the installation was completed.

maxwelldb reviewed (commented):
Left some comments, questions, and suggestions. Nothing major!

maxwelldb commented on modules/gitops-installing-argo-rollouts-cli-by-using-linux.adoc:
```diff
@@ -0,0 +1,62 @@
+[source,terminal]
+----
+# mv ./kubectl-argo-rollouts-linux-amd64 /usr/local/bin/kubectl-argo-rollouts
+----
+
+[IMPORTANT]
+====
```
```suggestion
----
+
[IMPORTANT]
====
```

Dhruv-Soni11 replied:
Fixed.

Dhruv-Soni11 commented on modules/gitops-argo-rollouts-cli-overview.adoc:
```diff
@@ -0,0 +1,17 @@
+[id="gitops-argo-rollouts-cli-overview_{context}"]
+= Argo Rollouts CLI overview
+
+Argo Rollouts CLI is an optional plugin that allows you to handle and monitor Argo Rollouts resources directly through the command line interface instead of using {OCP} web console or the `oc` command. 
```
Hi @maxwelldb - Yes, it is an optional plugin in this instance and needs to be stated explicitly somewhere as per SME's instruction. So I am retaining that part in the sentence and including the remaining part of the feedback too.

Dhruv-Soni11 commented on modules/gitops-argo-rollouts-cli-overview.adoc:
```diff
@@ -0,0 +1,17 @@
+
+Argo Rollouts CLI is an optional plugin that allows you to handle and monitor Argo Rollouts resources directly through the command line interface instead of using {OCP} web console or the `oc` command. 
+
+The Argo Rollouts CLI plugin enables you to perform many actions, including the following actions:
```
@maxwelldb, can we keep it as follows instead of your suggestion? 

The Argo Rollouts CLI plugin enables you to perform many actions, including the following:

maxwelldb replied:
It's less redundant, but even my suggestions wasn't perfect with respect to ISG. Strictly speaking, we should never be using "the following" as an adjective.

Dhruv-Soni11 replied:
So @maxwelldb, shall I remove the adjective and just keep the sentence as follows:

 The Argo Rollouts CLI plugin enables you to perform many actions, including:

maxwelldb replied:
Your original is correct per ISG. It uses "actions" redundantly, but it's the correct choice for translation. 

Other note, since I'm back in here:

⚠️ **enable** Use to describe the action of making an interface control or product function available to users. Avoid stating that inanimate objects grant abilities to people, as in “the product enables you to…”. Whenever possible, use a direct, user-focused alternative such as “you can use the product to …” or “with this product, users can…”.

So, all together, maybe:

```
With the Argo Rollouts CLI plugin, you can perform the following actions:
...

For a complete list of Argo Rollouts CLI plugin actions [commands?], see ...
```

I might even remove that list entirely if I'm in your shoes, as "manage and monitor" kind of gets at all of these things already, and the list is non-exhaustive, anyway. 

Ultimately, up to you! Peer review is advisory.

Dhruv-Soni11 replied:
Hi @maxwelldb - Thanks for the continued interest in improving the content further, I would like to stick to your next piece of suggestion for improving the lead-in sentence. However, there is no additional section that can provide information for a detailed list of Argo Rollouts CLI plugin actions so I am not including that line now.

Dhruv-Soni11 commented:
Hello @maxwelldb - I have incorporated the feedback you provided and request you to view my responses to your questions.

maxwelldb commented:
Setting this to peer-review-needed. I'm not on peer review squad this week. :(

maxwelldb commented on modules/gitops-argo-rollouts-cli-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="gitops-argo-rollouts-cli-overview_{context}"]
+= Argo Rollouts CLI overview
+
+You can use the Argo Rollouts CLI, which is an optional plugin, to manage and monitor Argo Rollouts resources directly, bypassing the need to use the {OCP} web console or the {oc-first}.
```
`oc-first` isn't rendering in the preview. Verify this before merge.

Dhruv-Soni11 replied:
Yeah, this attribute is not merging because it is not defined in the attributes file used in GitOps. Can we stick to {`oc`}???

Dhruv-Soni11 replied:
Keeping it as {oc} until advised otherwise.

Dhruv-Soni11 commented:
/retest

Dhruv-Soni11 commented:
/label merge-review-needed

skopacz1 commented on modules/gitops-argo-rollouts-cli-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="gitops-argo-rollouts-cli-overview_{context}"]
+= Argo Rollouts CLI overview
+
+You can use the Argo Rollouts CLI, which is an optional plugin, to manage and monitor Argo Rollouts resources directly, bypassing the need to use the {OCP} web console or the (`oc`).
```
I think the end of this sentence is missing CLI:
```suggestion
You can use the Argo Rollouts CLI, which is an optional plugin, to manage and monitor Argo Rollouts resources directly, bypassing the need to use the {OCP} web console or the CLI (`oc`).
```

Dhruv-Soni11 replied:
Fixed.

skopacz1 reviewed (commented):
Just one tiny nit, otherwise great job. 

Please note that you also have a conflict preventing me from merging this PR. So in addition to addressing my one comment, please also rebase the PR and resolve all conflicts. Thanks!

Dhruv-Soni11 commented:
> Just one tiny nit, otherwise great job.
> 
> Please note that you also have a conflict preventing me from merging this PR. So in addition to addressing my one comment, please also rebase the PR and resolve all conflicts. Thanks!

Hi @skopacz1 - Thank you for quickly reviewing my PR and providing feedback. I have incorporated your comment, rebased the PR, and resolved all the conflicts. I hope that the PR is now in a merge-able state. PTAL.

skopacz1 commented:
/cherrypick gitops-docs-1.13
